include/verilated.cpp

```diff
@@ -303,16 +303,18 @@ VlRNG::VlRNG() VL_MT_SAFE {
                         ^ (fromr.m_state[1] << 14));
     fromr.m_state[1] = (fromr.m_state[1] << 36) | (fromr.m_state[1] >> 28);
 }
-uint64_t VlRNG::rand64() VL_MT_UNSAFE {
+uint64_t VlRNG::rand64(bool reset, uint64_t salt) VL_MT_UNSAFE {
+    if (reset) { return (m_state[0] + m_state[1]) ^ salt; }
```

Comment:
This appears a misuse of VlRNG::rand64.  If you want a new seed you should be using srandom(...) instead.  I presume this is because you used VL_RANDOM, I think instead we should remove use of VL_RANDOM and make a new function.

This will also be faster as existing VL_RANDOM outside of init won't need to initialize and hit the new branch you added here.

Reply:
Good point, I'll loose the `reset` flag and just make a different method.  But do be clear, I'm not trying to set the seed here, I'm trying to piggy back on the seed that was set and get a psuedo-random value that is a function of the seed and the variable (by way of `salt`) without disturbing the state of `VlRNG`.

Reply:
Ah, ok.  I think we should rely on only the original seed and hash of signal hierarchy.

Another thing to add to test is inset a new signal first in the second test.